File: src/modules/BillingModule.js
/**
 * BillingModule.js
 * Responsibility: Calculate the final total including discounts and taxes.
 */
export const BillingModule = {
  calculateFinalTotal(subtotal, gst, discountPercent = 0) {
    const discountAmount = subtotal * (discountPercent / 100);
    const finalTotal = subtotal - discountAmount + gst;
    return {
      discountAmount,
      finalTotal
    };
  }
};

File: backend/BillingModule.cpp
#include "BillingModule.h"

BillingResult BillingModule::calculateFinalTotal(double subtotal, double gst, double discountPercent) {
    double discountAmount = subtotal * (discountPercent / 100.0);
    double finalTotal = subtotal - discountAmount + gst;
    return {discountAmount, finalTotal};
}
